### DISCUSSION TITLE
Not able to access env(DATABASE_URL) on Vercel deployment

### DISCUSSION THREAD

**Discussion by nmfrankel** [prisma/prisma#15238] [Q&A] [answered]
I'm working on [an open-source chat app](https://github.com/nmfrankel/ChatService) which is [deployed](https://chat-service-weld.vercel.app/) through [Vercel](https://vercel.com). Locally [Prisma](https://prisma.io) is working and able to interact to my [hosted cockroach DB](https://www.cockroachlabs.com/).

The problem is when the project gets built online the ``env(DATABASE_URL)`` is not getting read (I entered the env variable on the Vercel UI), thereby not letting me access my db. If this wasn't a public project I would just write the DATABASE_URL into the ``prisma/schema.prisma`` file, but in this case it's not an option.

Note: The ``/frontend`` Svelte-kit folder is the hosted folder, it's also the backend using Svelte-kit's ``+server.ts`` files located at ``/frontend/routes/api/..``.

Any help would be appreciated and if any further info is needed please let me know.

| Resource | Address |
|-----------|:----------:|
| Github Repo |  https://github.com/nmfrankel/ChatService/tree/bd0434882841af7ea3c4b5d14027c309df8cea57 |
| Deployment | https://chatservice-fc4ddavy7-nmfrankel.vercel.app/api/messages?id=1 |
| Page loading data | https://chatservice-fc4ddavy7-nmfrankel.vercel.app/threads (look at the console) | 

-    Links edited to be able to recreate

**Comment by nmfrankel** [comment, marked as answer]
Problem solved by the following two script and after three days of googling work. I feel this should be documented somewhere because whatever there are old and/or do not work.
```
"scripts": {
		...
		"vercel-build": "npx prisma generate && npm run build",
		"vercel-postbuild": "cp node_modules/@prisma/engines/*query* .vercel_build_output/functions/node/render/;cp prisma/schema.prisma .vercel_build_output/functions/node/render/"
}
```

### Credits
[Scott Spence](https://scottspence.com/posts/prisma-sveltekit-planetscale-deployment-issues#scripts)
[Mike Nikles](https://github.com/mikenikles/sveltekit-prisma/blob/main/package.json)

**Reply by nmfrankel** [reply]
This is only a fix for hosted databases, if you'd like to use ``env(DATABASE_URL)`` this will not resolve that issue.

**Reply by janpio** [reply]
What do you mean by "hosted databases" vs `env(DATABASE_URL)`?

**Reply by nmfrankel** [reply]
Thank you for reaching out at this point it's all working.

Hosted compared to local as in `file:./dev.db`, I understand Vercel can't doesn't offer persistent storage. It's all great 👍🏻
